gunpowder/nodes/csv_points_source.py

```diff
@@ -92,36 +94,33 @@ def provide(self, request):
 
     def _get_points(self, point_filter):
 
-        filtered = self.data[point_filter]
-        ids = np.arange(len(self.data))[point_filter]
+        if self.id_dim is not None:
+            filtered = self.data[point_filter][:,:self.ndims]
```

Comment:
is this assuming the `id_dim` is the last dimension? i.e. x, y, z, id. This looks like it would break if you had `id_dim=0`
It would also be nice to document the extra parameters in the class initialization

Reply:
I think you are correct, I'll update it.

Reply:
I thought about it. You are right that `id_dim` has to come after `:ndims` (doesn't have to be last, but has to come after it), but that is in line with the previous state where `:ndims` are the first `ndims` values if `ndims` is positive or all except the last `ndims` values if `ndims` is negative. Nevertheless, the point coordinates have to come first. However, there can still be more values per line that are ignored.

Reply:
Sounds good. Would you mind documenting those new arguments `__init__` docstring? Then I'll merge this

Reply:
updated the docstring